modules/blocks.php

```diff
@@ -6,77 +6,82 @@
  *
  */
 
-jetpack_register_block(
-	'map',
-	array(
-		'render_callback' => 'jetpack_map_block_load_assets',
-	)
-);
 
-jetpack_register_block( 'vr' );
-
-/**
- * Tiled Gallery block. Depends on the Photon module.
- *
- * @since 6.9.0
-*/
-if (
-	( defined( 'IS_WPCOM' ) && IS_WPCOM ) ||
-	class_exists( 'Jetpack_Photon' ) && Jetpack::is_module_active( 'photon' )
-) {
-	jetpack_register_block(
-		'tiled-gallery',
+if ( jetpack_register_gutenberg_extension( 'map' ) ) {
+	register_block_type(
+		'jetpack/map',
 		array(
-			'render_callback' => 'jetpack_tiled_gallery_load_block_assets',
+			'render_callback' => 'jetpack_map_block_load_assets',
 		)
 	);
 
 	/**
-	 * Tiled gallery block registration/dependency declaration.
+	 * Map block registration/dependency declaration.
 	 *
-	 * @param array  $attr - Array containing the block attributes.
-	 * @param string $content - String containing the block content.
+	 * @param array  $attr - Array containing the map block attributes.
+	 * @param string $content - String containing the map block content.
 	 *
 	 * @return string
 	 */
-	function jetpack_tiled_gallery_load_block_assets( $attr, $content ) {
+	function jetpack_map_block_load_assets( $attr, $content ) {
 		$dependencies = array(
 			'lodash',
+			'wp-element',
 			'wp-i18n',
-			'wp-token-list',
 		);
-		Jetpack_Gutenberg::load_assets_as_required( 'tiled-gallery', $dependencies );
 
-		/**
-		 * Filter the output of the Tiled Galleries content.
-		 *
-		 * @module tiled-gallery
-		 *
-		 * @since 6.9.0
-		 *
-		 * @param string $content Tiled Gallery block content.
-		 */
-		return apply_filters( 'jetpack_tiled_galleries_block_content', $content );
+		$api_key = Jetpack_Options::get_option( 'mapbox_api_key' );
+
+		Jetpack_Gutenberg::load_assets_as_required( 'map', $dependencies );
+		return preg_replace( '/<div /', '<div data-api-key="'. esc_attr( $api_key ) .'" ', $content, 1 );
 	}
 }
 
+jetpack_register_gutenberg_extension( 'vr' );
+
 /**
- * Map block registration/dependency declaration.
- *
- * @param array  $attr - Array containing the map block attributes.
- * @param string $content - String containing the map block content.
+ * Tiled Gallery block. Depends on the Photon module.
  *
- * @return string
- */
-function jetpack_map_block_load_assets( $attr, $content ) {
-	$dependencies = array(
-		'lodash',
-		'wp-element',
-		'wp-i18n',
-	);
+ * @since 6.9.0
+*/
+if (
+	( defined( 'IS_WPCOM' ) && IS_WPCOM ) ||
+	class_exists( 'Jetpack_Photon' ) && Jetpack::is_module_active( 'photon' )
+) {
+	if ( jetpack_register_gutenberg_extension( 'tiled-gallery' ) ) {
```

Comment:
Why not have this if clause be part of the same conditional above so that you don't need additional nesting?